src/attribution/onINP.ts

```diff
@@ -25,38 +26,102 @@ import {
   ReportOpts,
 } from '../types.js';
 
+// The maximum number of LoAF entries with interactions to keep in memory.
+// 10 is chosen here because it corresponds to the maximum number of
+// long interactions needed to compute INP, so no matter which one of those
+// interactions ends up being the INP candidate, it should correspond to one
+// of the 10 longest LoAF entries containing an interaction.
+const MAX_LOAFS_TO_CONSIDER = 10;
+
+// A list of longest LoAFs with interactions on the page sorted so the
+// longest one is first. The list is at most MAX_LOAFS_TO_CONSIDER long.
+const longestLoAFsList: PerformanceLongAnimationFrameTiming[] = [];
+
+// A PerformanceObserver, observing new `long-animation-frame` entries.
+// If this variable is defined it means the browser supports LoAF.
+let loafObserver: PerformanceObserver | undefined;
+
+const handleEntries = (entries: PerformanceLongAnimationFrameTiming[]) => {
+  entries.forEach((entry) => {
+    if (entry.firstUIEventTimestamp > 0) {
```

Comment:
I want to confirm that this is indeed the right way to limit LoAF collection to just those that contain a potential INP candidate.

Reply:
This might also discard input delays. Like if the UI event is handled after the LoAF

Reply:
Wouldn't the UI event result in extending the current frame normally in most cases as it's high priority? Or are you talking about if UI event happens during layout/paint on previous frame so has to wait until next frame? In that case the `firstUIEventTimestamp` would be in the previous LoAF so it wouldn't be counted?

I think there's a question of how much we're willing to accept these sorts of outliers versus holding a lot more potential LoAFs in memory? But yeah, these are why I didn't wanna manage a buffer of entries myself and had hoped to use the browser's buffer (before I realised it wasn't a rolling buffer and only first 200 entries).

Reply:
I've updated the implementation to check for LoAF intersection with Event Timing entries, so I'm going to resolve this.